### DISCUSSION TITLE
all in one vpn manager

### DISCUSSION THREAD

**Discussion by gabbifederico** [BornToBeRoot/NETworkManager#2164] [Ideas]
hi everyone
i usually use various vpn software
would't be nice if networkmanager could manage VPNs (fortinet client, pulse secure,cisco anyconnect) with external packages/Addons?

**Comment by BornToBeRoot** [comment]
Hi @gabbifederico,

thank you for the suggestion. I see several problems here:

1. Partially properitäre software if no vpn standards are used (not sure if i can use / implement / distribute them)
2. I do not have any of the above systems to test the implementation (in all its variations - authentication, encryption, etc.). The implementation and testing would also be extremely time consuming.
3. There are like hundreds of different VPN providers (Fortinet, Pulse Secure, Cisco, Ubiquiti, Palo Alto, WatchGuard, Check Point, Sophos, Barracuda, Sonic Wall, etc.) and many more Software VPNs (like Proton, NordVPN, etc.). 
4. Standards like OpenVPN and Wireguard might be possible. But I see other problems here:
    - Secure implementation
    - Security updates from a manufacturer -> I may have to provide an update for NETworkManager

I believe that the manufacturers can provide the clients better and more securely. Furthermore, I don't know how many here use multiple VPN clients on a daily basis and if there is really an advantage with this feature.
